tests/flask_stub_server.py

```diff
@@ -0,0 +1,150 @@
+import threading
```

Comment:
Are you planning to move the other tests to use flask?
Is the plan to add all the test cases to this shared stub server file? We will need to be careful to ensure this doesn't become a convoluted mess.

For instance, we have specific test setup for corner cases like these: https://github.com/Kaggle/kagglehub/blob/9c25bbfe2bcfcce847cbb5ddbe080b607f63d153/tests/test_kaggle_api_client.py#L22-L35

Reply:
As discussed offline, I will move all the tests over to flask. I will do a follow up PR.